src/extension.ts

```diff
@@ -407,6 +408,53 @@ function installCustomRequestHandlers(context: VSCode.ExtensionContext) {
   languageClient.onRequest(protocol.ShowTaintVulnerabilityRequest.type, showAllLocations);
 }
 
+async function isFileIgnoredForFolder(workspaceFolderPath: string, gitCommand: string): Promise<boolean> {
+  const {sout, serr} = await new Promise<{ sout: string, serr: string }>((resolve, reject) => {
+    ChildProcess.exec(
+        gitCommand,
+        {cwd: workspaceFolderPath},
+        (error: Error & { code?: 0 | 1 | 128 }, stdout, stderr) => {
+          if (error && (error.code !== 0 && error.code !== 1)) {
+            logToSonarLintOutput('Error on git gitCommand: ' + error);
+            reject(error);
+            return;
+          }
+          resolve({sout: stdout, serr: stderr});
+        }
+    );
+  });
+
+  if (serr) {
+    throw new Error(serr);
+  }
+
+  return Promise.resolve(sout.length > 0);
+}
+
+async function isIgnoredByScm(fileUri: string): Promise<boolean> {
+  const uriRelative = VSCode.workspace.asRelativePath(fileUri, false);
```

Comment:
The doc of this method says it will return the file itself if there is no workspace folders or if the file is not part of any of them.